Source/WebCore/editing/TextManipulationController.cpp

```diff
@@ -337,7 +337,7 @@ static bool isEnclosingItemBoundaryElement(const Element& element)
         }
 
         // Treat a or li with white-space: nowrap as its own paragraph so that wrapping whitespace between them will be preserved.
-        if (renderer->style().whiteSpace() == WhiteSpace::NoWrap)
+        if (renderer->style().textWrap() == TextWrap::NoWrap && renderer->style().whiteSpaceCollapse() == WhiteSpaceCollapse::Collapse)
```

Comment:
@whsieh do you know if we actually care about whitespace collapse here? Replacing white-space with text-wrap/collapse in many cases is not a mechanical change but rather a chance to drop one of the checks.
@nt1m please either remove or update comment above.

Reply:
I reviewed this on the assumption that there would be no behavior change, based on Tim's commit message (as well as the fact that the API test `TextManipulation.CompleteTextManipulationPreservesWhitespacesBetweenItems` still passes).

It would help to test the original use case (webpage translation in https://www.ukr.net/ua/), and verify that that bug remains fixed.

Reply:
It might be possible to drop the `white-space-collapse` check based on the comment, given it seems like `white-space-collapse: preserve` would preserve the wrapping whitespace anyway? But since it shouldn't make a difference, I decided to stay on the safe side.

Reply:
I think reducing code complexity wins over staying on the safe side, when we don't have to rush the change in.

Reply:
OK, I'll give it a shot then.